pkg/controller/controller.go

```diff
@@ -191,25 +196,43 @@ func (c *Controller) onDelete(obj interface{}) {
 	}
 }
 
-func (c *Controller) onAddActionSet(as *crv1alpha1.ActionSet) error {
-	as, err := c.crClient.CrV1alpha1().ActionSets(as.GetNamespace()).Get(context.TODO(), as.GetName(), v1.GetOptions{})
+func (c *Controller) createBpCache(ctx context.Context, ns string) (map[string]*crv1alpha1.Blueprint, error) {
+	bps, err := c.crClient.CrV1alpha1().Blueprints(ns).List(ctx, v1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	bpMap := make(map[string]*crv1alpha1.Blueprint)
+	for _, bp := range bps.Items {
+		bpMap[bp.Name] = bp
+	}
+
+	return bpMap, nil
+}
+
+func (c *Controller) onAddActionSet(ctx context.Context, t *tomb.Tomb, as *crv1alpha1.ActionSet) error {
+	as, err := c.crClient.CrV1alpha1().ActionSets(as.GetNamespace()).Get(ctx, as.GetName(), v1.GetOptions{})
 	if err != nil {
 		return errors.WithStack(err)
 	}
 	if err := validate.ActionSet(as); err != nil {
 		return err
 	}
+	bps, err := c.createBpCache(ctx, as.GetNamespace())
```

Comment:
This will get expensive during a controller restart or if the actionset get re-queued. See https://github.com/kanisterio/kanister/pull/1267. IIRC, on restart, the controller will call `onAddActionSet()` on all existing actionset resources. With this change, the controller will create a map of _all_ blueprints for _every_  known actionset.